### PR TITLE
Multilevel doubly linked list

### PR DESCRIPTION
**Pull request by jquigtar**
The implementation of flatten is correct but a little shaky on general driver implementation to show how it works. Also needed some time to get used to using GitHub.

### CODE REVIEW

File: 2020-01-Winter/1_linked_lists/Flatten_Multilevel_Doubly_Linked_List/LinkedList.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+		/* Find tail node of first level linked list */
+		Node tail = head; 
+		while (tail.next != null) { 
+			tail = tail.next; 
+		} 
+
+		// One by one traverse through all nodes of first level 
+		// linked list till we reach the tail node 
+		Node current = head; 
+		while (current != tail) {
```
Comment (etcadinfinitum) [inline 2020-01-Winter/1_linked_lists/Flatten_Multilevel_Doubly_Linked_List/LinkedList.java]: This is going to produce a fencepost error if the first list's tail has its own child list(s); fix this by making the following changes:
1. this line should read `while (current != nullptr) {`
2. line 156 (`current.next.prev = temp;`) should be wrapped in an if statement checking that `current.next` is not null

You could add the pointer manipulation operations after the while block to fix the fencepost error; the changes suggested above might be cleaner.

File: 2020-01-Winter/1_linked_lists/README.md [hunk 2]
```diff
@@ -118,15 +159,50 @@ Go to [Top](#top)
...
+				current.child = null;
+			} 
+
+			// Change current node 
+			current = current.next; 
+		} 
+		return head;
+	}
 
 #### Testing The Solutions OR Driver For Solution
```
Comment (etcadinfinitum) [inline 2020-01-Winter/1_linked_lists/README.md]: Add a link here to the Java file you wrote, or just remove this section altogether

**Comment by jquigtar** [comment]
I have fixed the edits you suggested. let me know if there's anything else to fix

File: 2020-01-Winter/1_linked_lists/README.md [hunk 3]
```diff
@@ -118,19 +159,58 @@ Go to [Top](#top)
...
+		/* Find tail node of first level linked list */
+		Node tail = head; 
+		while (tail.next != null) { 
+			tail = tail.next; 
+		} 
+
+		// One by one traverse through all nodes of first level 
+		// linked list till we reach the tail node 
+		Node current = head; 
+		while (current != tail) {
```
Comment (etcadinfinitum) [inline 2020-01-Winter/1_linked_lists/README.md]: Fixing writeup version of solution method to align with source file
```suggestion
		while (current != null) { 
```

File: 2020-01-Winter/1_linked_lists/README.md [hunk 4]
```diff
@@ -118,19 +159,58 @@ Go to [Top](#top)
...
+				// then link child in between current and current.next
+				//store last link on next level as temp to connect to
+				//current.next
+				Node temp = current.child; 
+				while (temp.next != null) { 
+					temp = temp.next; 
+				} 
+				//break and create new links
+				temp.next = current.next;
+				current.next.prev = temp;
```
Comment (etcadinfinitum) [inline 2020-01-Winter/1_linked_lists/README.md]:
```suggestion
				if(current.next != null){
					current.next.prev = temp;
				}
```

File: 2020-01-Winter/1_linked_lists/README.md [hunk 5]
```diff
@@ -118,19 +159,58 @@ Go to [Top](#top)
...
+			// Change current node 
+			current = current.next; 
+		} 
+		return head;
+	}
+```
+
+#### Driver For Solution
+
+[Driver](CTCI/2020-01-Winter/1_linked_lists/Flatten_Multilevel_Doubly_Linked_List/LinkeList.java)
```
Comment (etcadinfinitum) [inline 2020-01-Winter/1_linked_lists/README.md]: 404 :x: 
```suggestion
[Driver](CTCI/2020-01-Winter/1_linked_lists/Flatten_Multilevel_Doubly_Linked_List/LinkedList.java)
```

**Pull request merged**